remotes/handlers.go

```diff
@@ -215,3 +218,55 @@ func push(ctx context.Context, provider content.Provider, pusher Pusher, desc oc
 	rd := io.NewSectionReader(ra, 0, desc.Size)
 	return content.Copy(ctx, cw, rd, desc.Size, desc.Digest)
 }
+
+// PushContent pushes content specified by the descriptor from the provider.
+//
+// Base handlers can be provided which will be called before any push specific
+// handlers.
+func PushContent(ctx context.Context, provider content.Provider, pusher Pusher, baseHandlers []images.Handler, desc ocispec.Descriptor) error {
```

Comment:
We generally use destination first argument order, so this should be something like this:

```
PushContent(ctx context.Context, pusher Pusher, desc ocispec.Descriptor, provider content.Provider, baseHandlers []images.Handler)
```

The `baseHandlers` may make sense as variadic.

[`PushHandler`](https://godoc.org/github.com/containerd/containerd/remotes#PushHandler) needs to be fixed to follow this convention, as well.